### PR TITLE
Partially use ``namespace`` in ``PyLinter``

### PR DESCRIPTION
**Pull request by DanielNoord**
- [x] Add yourself to CONTRIBUTORS if you are a new contributor.
- [x] Write a good description on what the PR does.

## Type of Changes

|     | Type                   |
| --- | ---------------------- |
| ✓   | :hammer: Refactoring   |

## Description

These are the easy options without complicated test that need updating.

### CODE REVIEW

File: pylint/lint/pylinter.py [hunk 1]
```diff
@@ -1442,7 +1444,10 @@ def _get_message_state_scope(
         """Returns the scope at which a message was enabled/disabled."""
         if confidence is None:
             confidence = interfaces.UNDEFINED
-        if self.config.confidence and confidence.name not in self.config.confidence:
+        if (
+            self.namespace.confidence
+            and confidence.name not in self.namespace.confidence
+        ):
             return MSG_STATE_CONFIDENCE  # type: ignore[return-value] # mypy does not infer Literal correctly
```
Comment (Pierre-Sassoulas) [inline pylint/lint/pylinter.py:1451]: Look like we handling the case where ``self.config.confidence`` is falsey, but it should be None or a confidence, at this point, right ? Is that for mypy ?

File: tests/lint/unittest_lint.py [hunk 2]
```diff
@@ -239,7 +240,7 @@ def test_enable_message_category(initialized_linter: PyLinter) -> None:
 
 
 def test_message_state_scope(initialized_linter: PyLinter) -> None:
-    class FakeConfig:
+    class FakeConfig(argparse.Namespace):
```
Comment (Pierre-Sassoulas) [inline tests/lint/unittest_lint.py:243]: 👍

File: pylint/lint/pylinter.py (hunk 1 above)
Comment (DanielNoord) [reply pylint/lint/pylinter.py:1451]: That's interesting actually. I'll try and see if this could just be an `in` check.

**Comment by DanielNoord** [comment]
Drafting while I figure out why this decreases coverage. Man this morning is rough...

**Comment by DanielNoord** [comment]
@Pierre-Sassoulas PR became a bit different than original title but this doesn't decrease coverage :) Uncovered lines are already uncovered.

**Review by Pierre-Sassoulas — APPROVED** [review]
👍

**Pull request merged**